src/browser/components/pages/TemplateTask/index.js

```diff
@@ -0,0 +1,130 @@
+import React, { Component } from 'react'
+import TemplateTaskForm from '../../molecules/TemplateTaskForm'
+import List from '../../atoms/List'
+import Button from '../../atoms/Button'
+// import styles from './index.css'
+
+export default class TemplateTask extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      showTemplateTaskForNoob: false,
+      showTemplateTaskForMentor: false,
+      currentTemplateTasks: { noob:[],mentor:[] },
+      noobTempTaskFormData: {name:'', desc:'', daysToComplete:''},
+      mentorTempTaskFormData: {name:'', desc:'', daysToComplete:''}
+    }
+    this.addTemplateTask = this.addTemplateTask.bind(this)
+    this.updateTemplateTask = this.updateTemplateTask.bind(this)
+  }
+
+  getTemplateTask() {
+    fetch('/api/template_tasks', {
+      method: 'get',
+      mode: 'cors',
+      credentials: 'same-origin'
+    }).then( results => {
+      return results.json()
+    }).then(tasks => {
+      this.setState({
+        currentTemplateTasks : tasks
+      })
+    })
+  }
+
+  addTemplateTask(event, role){
+    const newTemplateTask = this.props.location.query
+    newTemplateTask.role = role
+    event.preventDefault()
+    fetch('/api/template_tasks', {
+      method: 'post',
+      mode: 'cors',
+      credentials: 'same-origin',
+      body: JSON.stringify(newTemplateTask),
+      headers: new Headers({
+        'Content-Type': 'application/json',
+        'Accept': 'application/json'
+      })
+    }).then(res => {
+    }).then( newTask => {
+      this.getTemplateTask()
+      this.setState({
+        showTemplateTaskForNoob: false,
+        showTemplateTaskForMentor: false,
+      })
+    })
+  }
+
+  updateTemplateTask(event) {
+
+  }
+
+  deteleTemplateTask(event) {
+    // fetch('/api/template_tasks')
+  }
+
+
+  handleTempTaskFormFields(event, inputField) {
+    if(inputField === 'name'){
+      this.props.location.query.template_task_name = event.target.value
```

Comment:
why are you changing the `props` here? `this.props` is immutable, and should not be changed